### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by barchetta**
- Eliminate use of some deprecated methods
- MP: Use built-in io.helidon.microprofile.cdi.Main when custom Main not needed 
- MP: remove Application classes when not needed
- Remove reference to JDK8 from READMEs
- Other fixes

### CODE REVIEW

File: examples/microprofile/hello-world-explicit/src/main/java/io/helidon/microprofile/example/helloworld/explicit/Main.java [hunk 1]
```diff
@@ -34,7 +34,6 @@ private Main() {
      */
     public static void main(String[] args) {
         Server server = Server.builder()
-                .addApplication(HelloWorldApplication.class)
                 // using a customized helidon config instance (in this case the default...)
```
Comment (tomas-langer) [inline examples/microprofile/hello-world-explicit/src/main/java/io/helidon/microprofile/example/helloworld/explicit/Main.java:37]: The message is wrong - this uses the MP Config, not Helidon config

File: examples/microprofile/mp1_1-static-content/src/main/java/io/helidon/microprofile/example/staticc/Main.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, 2020 Oracle and/or its affiliates.
```
Comment (tomas-langer) [inline examples/microprofile/mp1_1-static-content/src/main/java/io/helidon/microprofile/example/staticc/Main.java:2]: This main class seems to be useless.

Comment (tomas-langer) [reply examples/microprofile/mp1_1-static-content/src/main/java/io/helidon/microprofile/example/staticc/Main.java:2]: Also type in package name

File: examples/microprofile/hello-world-explicit/src/main/java/io/helidon/microprofile/example/helloworld/explicit/Main.java (hunk 1 above)
Comment (barchetta) [reply examples/microprofile/hello-world-explicit/src/main/java/io/helidon/microprofile/example/helloworld/explicit/Main.java]: Fixed

File: examples/microprofile/mp1_1-static-content/src/main/java/io/helidon/microprofile/example/staticc/Main.java (hunk 2 above)
Comment (barchetta) [reply examples/microprofile/mp1_1-static-content/src/main/java/io/helidon/microprofile/example/staticc/Main.java:2]: `staticc`? I think that is short for static content. I will leave this as-is for now.

File: examples/webserver/basics/src/main/java/io/helidon/webserver/examples/basics/NameReader.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public boolean accept(GenericType<?> type, MessageBodyReaderContext context) {
+        return context.contentType()
+                .map(ct -> MediaType.parse("application/name").equals(ct))
+                .map(acceptable -> acceptable && Name.class.isAssignableFrom(type.rawType()))
+                .orElse(false);
+    }
+}
+
+
```
Comment (romain-grecourt) [inline examples/webserver/basics/src/main/java/io/helidon/webserver/examples/basics/NameReader.java:52]: That's a lot of empty lines :)

**Pull request merged**